### DISCUSSION TITLE
Clarification of "Balancing Stability and Feature Development"

### DISCUSSION THREAD

**Discussion by VladimirFokow** [moveit/moveit2#2810] [Q&A]
Please, help me clarify some things about [Balancing Stability and Feature Development in Moveit and ROS](https://github.com/moveit/moveit2/discussions/2190), particularly about the section:

## "I want to use MoveIt from the main branch on ROS Humble but don't want to build it from the source"

1. I ran the following commands (according to [this README](https://github.com/moveit/moveit2_packages/blob/jammy-humble/README.md), and [this page](https://moveit.ros.org/install-moveit2/binary/))

    ```bash
    echo "deb [trusted=yes] https://github.com/moveit/moveit2_packages/raw/jammy-humble/ ./" | sudo tee /etc/apt/sources.list.d/moveit_moveit2_packages.list
    echo "yaml https://github.com/moveit/moveit2_packages/raw/jammy-humble/local.yaml humble" | sudo tee /etc/ros/rosdep/sources.list.d/1-moveit_moveit2_packages.list

    sudo apt update && sudo apt upgrade && sudo apt install -y ros-humble-moveit
    ```

    Then I tried to follow the [introductory tutorials](https://moveit.picknik.ai/main/doc/tutorials/quickstart_in_rviz/quickstart_in_rviz_tutorial.html), but it looks like the package `moveit2_tutorials` isn't installed. Also it doesn't find [`moveit_servo`](https://moveit.picknik.ai/main/doc/examples/realtime_servo/realtime_servo_tutorial.html#using-the-c-api), and it looks like many other essential things are missing.

    Why aren't they available?
    How to make them available? - So that the binary installation isn't different from the building from source instructions. 

<br />

2.  Is this correct? If I want to install Moveit2 to use with ROS2 Humble, I currently have 3 (or more?) options:

    1. follow [these instructions](https://moveit.picknik.ai/humble/doc/tutorials/getting_started/getting_started.html) (build Moveit2-humble from source - which is a stable but outdated version of Moveit2)
    2. follow [these instructions](https://moveit.picknik.ai/main/doc/tutorials/getting_started/getting_started.html) (build Moveit2-main from source - which is literally the main branch of Moveit2 github repo)
    3. follow instructions I wrote above - from the discussion  
    (binary install of Moveit2. It's faster because we don't have to build it, but is unstable, and doesn't correspond to docs)
    4. Is there an option to `apt install` the stable `moveit2-humble`? To do this, would it just be enough to not run the above `echo` lines?
    
    About option (iii.): it installs not a `humble` branch of moveit2, but a very outdated `main` branch (as of many months ago), which may be outdated to the current main docs online; so it doesn't have actual corresponding docs to that specific version of main. Is it correct?

    For information, when installed with option (iii.),
    this hints that the installed version is from release "2.7.4":
    ```bash
    $ apt policy ros-humble-moveit
    ros-humble-moveit:
    Installed: 2.7.4-42-g92ca89dca-2023.08.10.02.16
    Candidate: 2.7.4-42-g92ca89dca-2023.08.10.02.16
    Version table:
    *** 2.7.4-42-g92ca89dca-2023.08.10.02.16 500
            500 https://github.com/moveit/moveit2_packages/raw/jammy-humble ./ Packages
            100 /var/lib/dpkg/status
        2.5.5-1jammy.20240404.154053 500
            500 http://packages.ros.org/ros2/ubuntu jammy/main amd64 Packages
    ```
    but currently, the [latest release](https://github.com/ros-planning/moveit2/tags) is 2.9.0. What am I misunderstanding? 

    Is it possible to know the exact commit of moveit2 repository that is used when installed from binary?

<br />

3. The discussion says it is build "nightly":

    > Debians for MoveIt from the main branch on all supported versions of ROS that is built nightly in GitHub Actions.

    Does "nightly" mean every 24 hours?
    I'm confused. The last commit and executed action workflows [in that repo](https://github.com/moveit/moveit2_packages/actions) were 8 months ago. 
    Is this information outdated? 
    Is that repo abandoned?

<br />

4. Can these instructions and explanations be in the [getting started guide](https://moveit.picknik.ai/humble/doc/tutorials/getting_started/getting_started.html) in each docs version where they apply: humble, iron, rolling, etc?
    (to install MoveIt from binary so that it works with ROS2 Humble, and people can run the tutorials with it, and if any differences vs building from source, they are well-described)

**Comment by tylerjw** [comment]
Rolling has been broken for a while and no one at PickNik seems to care about it. That workaround with the apt packages built in CI stopped working months ago and is not supported. 

The only supported workflow I know of for using the latest on main is to build from source.

**Reply by VladimirFokow** [reply]
Thank you for the update!
